src/Toolkit/OutBox/OutBoxStarter.cs

```diff
@@ -0,0 +1,111 @@
+        if (!Enum.TryParse(strDbType, out dbType))
+            throw new ArgumentNullException($"Invalid DbType ({strDbType}) informed on {_DbTypeVarName} variable. Unable to start Transactional OutBox.");
+        OutBoxDbContext.SetDbType(dbType);
+        var db = Enum.GetName(OutBoxDbContext.DbType);
+        if (_DbConnectionVarName.IsEmpty())
+            throw new ArgumentNullException($"{db} Connection variable name not provided. Unable to start consumer target host.");
+        var stringConnection = Environment.GetEnvironmentVariable(_DbConnectionVarName);
+        if (stringConnection.IsEmpty())
+            throw new ArgumentNullException($"Unable to identify {db} Connection on {_DbConnectionVarName} variable. Unable to start Transactional OutBox.");
+        DoUseDatabase(stringConnection);
+        return this;
+    }
+
+    public void UseRabbitMq(string rabbitMqVariableName = "RABBIT_MQ")
+    {
+        if (rabbitMqVariableName.IsEmpty())
+            throw new ArgumentNullException("RabbitMq variable name not provided. Unable to start consumer target host.");
+        string host = Environment.GetEnvironmentVariable(rabbitMqVariableName);
+        if (host.IsEmpty())
+            throw new ArgumentNullException($"Unable to identify RabbitMq Host on {rabbitMqVariableName} variable. Unable to start Transactional OutBox.");
+        DoUseRabbitMq(host);
+    }
+
+    public IOpenTelemetreable UseSerilog()
+    {
+        Log.Logger = new LoggerConfiguration()
+            .MinimumLevel.Information()
+            .MinimumLevel.Override("MassTransit", LogEventLevel.Debug)
+            .MinimumLevel.Override("Microsoft", LogEventLevel.Warning)
+            .MinimumLevel.Override("Microsoft.EntityFrameworkCore", LogEventLevel.Warning)
+            .MinimumLevel.Override("Microsoft.EntityFrameworkCore.Database.Command", LogEventLevel.Warning)
+            .Enrich.FromLogContext()
+            .WriteTo.Console()
+            .CreateLogger();
+        Builder.Host.UseSerilog();
+        return this;
+    }
+
+    public IDatabaseable UseOpenTelemetry()
+    {
+        Builder.Services.AddOpenTelemetryTracing(builder =>
+        {
+            builder.SetResourceBuilder(ResourceBuilder.CreateDefault()
+                    .AddService(TelemetryName)
+                    .AddTelemetrySdk()
+                    .AddEnvironmentVariableDetector())
+                .AddSource("MassTransit")
+                .AddAspNetCoreInstrumentation()
+                .AddJaegerExporter(o =>
+                {
+                    o.AgentHost = HostMetadataCache.IsRunningInContainer ? "jaeger" : "localhost";
+                    o.AgentPort = 6831;
+                    o.MaxPayloadSizeInBytes = 4096;
+                    o.ExportProcessorType = ExportProcessorType.Batch;
+                    o.BatchExportProcessorOptions = new BatchExportProcessorOptions<Activity>
+                    {
+                        MaxQueueSize = 2048,
```

Comment:
Essas configurações poderiam estar em variável de ambiente ou então poderíamos extrair essa implementação em um método e passar esses valores como parâmetros.

Reply:
Sim, poderiam sim, mas acredito que não serão mudadas com tanta facilidade, é só uma questão de aumento de complexidade que pode ser feita quando de fato for necessário.